src/Engine/GameEvents/BattleEvent.java

```diff
@@ -61,13 +61,13 @@ public void performEvent(MapMaster gameMap)
     // Apply the battle results that we calculated previously.
     Unit attacker = battleInfo.attacker;
     Unit defender = battleInfo.defender;
-    battleInfo.attackerWeapon.fire(); // expend ammo
+    battleInfo.attackerWeapon.fire(attacker); // expend ammo
```

Comment:
This syntax feels a bit convoluted. I might prefer `attacker.fire(battleInfo.attackerWeapon)`.